bin/plugin/commands/changelog.js

```diff
@@ -538,6 +538,24 @@ function getEntry( issue ) {
 		: `- ${ title } ([${ issue.number }](${ issue.html_url }))`;
 }
 
+/**
+ * Returns a formatted changelog entry for a given issue object and matching feature name, or undefined
+ * if entry should be omitted.
+ *
+ * @param {IssuesListForRepoResponseItem} issue       Issue object.
+ * @param {string}                        featureName Feature name.
+ *
+ * @return {string=} Formatted changelog entry, or undefined to omit.
+ */
+function getFeatureEntry( issue, featureName ) {
+	return getEntry( issue )
+		?.replace(
+			new RegExp( `\\[${ featureName.toLowerCase() } \- `, 'i' ),
+			'['
+		)
+		.replace( new RegExp( `${ featureName.toLowerCase() }: `, 'i' ), '' );
```

Comment:
```suggestion
		.replace( new RegExp( `(?<=^- )${ featureName.toLowerCase() }: `, 'i' ), '' );
```
Adding this lookbehind to the expression would prevent [this strange output](https://github.com/WordPress/gutenberg/pull/34071#discussion_r688484038), as it will not replace the feature unless it's a full match.